pytorch_lightning/accelerators/accelerator.py

```diff
@@ -436,6 +438,10 @@ def restore_checkpoint_after_pre_dispatch(self) -> bool:
         """
         return self.training_type_plugin.restore_checkpoint_after_pre_dispatch
 
+    @abstractmethod
+    def get_device_stats(self, device: Union[str, torch.device]) -> Dict[str, Any]:
```

Comment:
What's the point of having this abstract of at least one subclass raises NotImplemented here? Can't we just raise it directly here and only overwrite it where we actually have an implementation?

I get that ideally we would have purely abstract base classes for interface, but personally I'm not a fan of mixing the concepts within a subpackage.

Reply:
That makes sense, I am fine with doing that. cc @ananthsub

Reply:
sounds good to me